src/scimlfunctions.jl

```diff
@@ -1993,48 +1993,44 @@ and more. For all cases, `u` is the state and `p` are the parameters.
-  This function should return a scalar.
-- `adtype`: see the section "Defining Optimization Functions via AD"
+- `f(u,p,args...)`: the function to optimize. `u` are the optimization variables and `p` are parameters used in definition of
+the objective, even if no such parameters are used in the objective it should be an argument in the function. This can also take
+any additonal arguments that are relevant to the objective function, for example minibatches used in machine learning,
+take a look at the minibatching tutorial [here](https://docs.sciml.ai/Optimization/stable/tutorials/minibatch/). This should return
+a scalar, the loss value, as the first return output and if any additional outputs are returned, they will be passed to the `callback`
+function described in [Callback Functions](@ref).
+- `adtype`: see the section [Defining Optimization Functions via AD](@ref)
 
 ## Keyword Arguments
 
-- `grad(G,u,p)` or `G=grad(u,p)`: the gradient of `f` with respect to `u`
-- `hess(H,u,p)` or `H=hess(u,p)`: the Hessian of `f` with respect to `u`
-- `hv(Hv,u,v,p)` or `Hv=hv(u,v,p)`: the Hessian-vector product ``(d^2 f / du^2) v``.
-- `cons(res,x,p)` or `cons(x,p)` : the constraints function, should mutate or return a vector
+- `grad(G,u,p)`: the gradient of `f` with respect to `u`. If `f` takes additional arguments
+    then `grad(G,u,p,args...)` should be used.
+- `hess(H,u,p)`: the Hessian of `f` with respect to `u`. If `f` takes additional arguments
+    then `hess(H,u,p,args...)` should be used.
+- `hv(Hv,u,v,p)`: the Hessian-vector product ``(d^2 f / du^2) v``. If `f` takes additional arguments
+    then `hv(Hv,u,v,p,args...)` should be used.
+- `cons(res,x,p)`: the constraints function, should mutate the passed `res` array
     with value of the `i`th constraint, evaluated at the current values of variables
     inside the optimization routine. This takes just the function evaluations
     and the equality or inequality assertion is applied by the solver based on the constraint
     bounds passed as `lcons` and `ucons` to [`OptimizationProblem`](@ref), in case of equality
     constraints `lcons` and `ucons` should be passed equal values.
-- `cons_j(res,x,p)` or `res=cons_j(x,p)`: the Jacobian of the constraints.
-- `cons_h(res,x,p)` or `res=cons_h(x,p)`: the Hessian of the constraints, provided as
-   an array of Hessians, with `res[i]` being the Hessian with respect to the `i`th output on `cons`.
-- `lag_h(res,x,sigma,mu,p)` or `res=lag_h(x,sigma,mu,p)`: the Hessian of the Lagrangian,
-    where `sigma` is a multiplier of the cost function and `mu` are the Lagrange multipliers
-    multiplying the constraints. This can be provided instead of `hess` and `cons_h`
-    to solvers that directly use the Hessian of the Lagrangian.
-- `paramjac(pJ,u,p)`: returns the parameter Jacobian ``df/dp``.
+- `cons_j(res,x,p)`: the Jacobian of the constraints.
+- `cons_h(res,x,p)`: the Hessian of the constraints, provided as
```

Comment:
missing the oop version

Reply:
The oop versions are not supported yet right

Reply:
No, that's a solver implementation thing, not an interface thing.

Reply:
There have been a couple of issues asking about this and since it is not there I think it makes sense to remove it for now and add back once it is implemented?